File: gang/week11/명예의 전당(1).java
import java.util.Arrays;

class Solution {
    public int[] solution(int k, int[] score) {
        int[] answer = new int[score.length];
        int[] rank = new int[k];
        rank[0] = score[0];
        int last = score[0];
        answer[0] = score[0];

        if (score.length <= k) {    // score 배열의 크기가 k 이하일 경우 모든 값이 명예의 전당에 오름
            for (int i = 1; i < score.length; i++) {    // 이 때 값을 비교하여 answer에 값 넣음
                rank[i] = score[i];
                if (score[i] > last) {
                    answer[i] = last;
                } else {
                    answer[i] = score[i];
                    last = score[i];
                }
            }
        } else {    // score 배열의 크기가 k보다 클 때
            for (int i = 1; i < k; i++) {   // k까지 명예에 전당에 올림
                rank[i] = score[i];
                if (score[i] > last) {
                    answer[i] = last;
                } else {
                    answer[i] = score[i];
                    last = score[i];
                }
            }

            Arrays.sort(rank);  // 명예의 전당 정렬

            for(int i = k; i < score.length; i++) {     // k이후부터 명예의 전당에서 제일 낮은 값과 비교
                if(score[i] > rank[0]) {
                    rank[0] = score[i];
                    Arrays.sort(rank);
                    answer[i] = rank[0];
                }
                else {
                    answer[i] = rank[0];
                }
            }
        }

        return answer;
    }
}

File: jeonghyeon/11week/명예의 전당(1).js
function solution(k, score) {
    var answer = []; // 결과
    var num = []; // 명예의 전당 들어있는 점수들
    for(let i = 0 ; i < score.length ; i ++) {
        if(i < k) {
            num.push(score[i]);
        }
        if(score[i]>Math.min(...num)) {
            num.pop(); // 제일 작은 수 빼기
            num.push(score[i]);
            num.sort((a,b) => b-a); // 내림차순 작은게 제일 뒤
        }
        answer.push(num.at(-1));
    }
    return answer;
}